exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -307,3 +315,235 @@ func (r *MachinePoolReconciler) reconcileInfrastructure(ctx context.Context, clu
 
 	return ctrl.Result{}, nil
 }
+
+// reconcileMachines reconciles Machines associated with a MachinePool.
+//
+// Note: In the case of MachinePools the machines are created in order to surface in CAPI what exists in the
+// infrastructure while instead on MachineDeployments, machines are created in CAPI first and then the
+// infrastructure is created accordingly.
+// Note: When supported by the cloud provider implementation of the MachinePool, machines will provide a means to interact
+// with the corresponding infrastructure (e.g. delete a specific machine in case MachineHealthCheck detects it is unhealthy).
```

Comment:
Q: Will the MHC support be implemented in a follow-up? (just to be clear, I'm absolutely in favor of a follow-up instead of including it in this PR)

Reply:
I'm not sure. IIRC I added this as a suggestion from @CecileRobertMichon who is OOF right now.

Reply:
Just to give some context. The MHC controller only adds a condition. The actual remediation of a Machine is implemented in the "owning" controller. This means today in the KCP and the MachineSet controller. Without explicitly implementing this, MHC won't work for MachinePool.

No rush on this, but we'll probably end up to create a follow-up issue to implement MHC support for MachinePools